### PR TITLE
bugfix: fixes #3732 (bug in compilation of inspect)

### LINKED ISSUE #3732: bug: compiler bug when using inspect system function with endpoint receiving tuple arguments
https://forum.dfinity.org/t/motoko-compiler-bug-when-using-inspect-system-function-with-endpoint-receiving-tuple-arguments/18187

Hello community! Today I found that I cannot compile my actor anymore after migrating my shared function to receive tuple of Nat-s instead of one single Nat. I narrowed down the issue and found that inspect function is to blame, it does not work when it’s declared

```
DFX version: 0.12.1 and 0.12.2-beta.0
OS: MacOS Ventura 13.1
```

The error:
```
Ill-typed intermediate code after Desugaring (use -v to see dumped IR):
(unknown location): IR type error [M0000], bad case
Raised at Ir_def__Check_ir.error.(fun) in file "ir_def/check_ir.ml", line 95, characters 30-92
Called from Stdlib__list.iter in file "list.ml", line 110, characters 12-15
Called from Ir_def__Check_ir.check_exp in file "ir_def/check_ir.ml", line 712, characters 4-30
Called from Ir_def__Check_ir.check_dec in file "ir_def/check_ir.ml", line 1084, characters 4-21
Called from Stdlib__list.iter in file "list.ml", line 110, characters 12-15
Called from Ir_def__Check_ir.check_exp in file "ir_def/check_ir.ml", line 695, characters 4-22
Called from Ir_def__Check_ir.check_dec in file "ir_def/check_ir.ml", line 1084, characters 4-21
Called from Stdlib__list.iter in file "list.ml", line 110, characters 12-15
Called from Ir_def__Check_ir.check_exp in file "ir_def/check_ir.ml", line 695, characters 4-22
Called from Ir_def__Check_ir.check_comp_unit in file "ir_def/check_ir.ml", line 1150, characters 4-27
Called from Ir_def__Check_ir.check_prog in file "ir_def/check_ir.ml", line 1163, characters 6-28
```
Minimal reproducing example:
```

actor class TestAPI() {
  public func endpoint(x: (Nat,Nat)) : async () {};
  system func inspect({}) : Bool = true;
};
```
Thanks for help and advises in advance!

### PR DESCRIPTION
**Pull request by crusso**
The compiler was incorrectly flattening a tuple returned by deserialization into multiple returns because it used lambda helper `-->` instead of `funcE`. The latter won't try to infer the return arity for you (`-->` gets it wrong in this case).

Fixes #3732.

### CODE REVIEW

File: test/run-drun/issue-3732.mo [hunk 1]
```diff
@@ -0,0 +1,4 @@
+actor class TestAPI() {
+  public func endpoint(x: (Nat,Nat)) : async () {};
+  system func inspect({}) : Bool = true;
```
Comment (nomeata) [inline test/run-drun/issue-3732.mo:3]: Do we need both for a repo?

Comment (crusso) [reply test/run-drun/issue-3732.mo]: No just one.

File: src/lowering/desugar.ml [hunk 2]
```diff
@@ -356,10 +356,10 @@ and call_system_func_opt name es obj_typ =
               (List.map (fun tf ->
                 (tf.T.lab,
                   match tf.T.typ with
-                  | T.Func(T.Local, _,  [], [], ts) ->
-                    let unit = fresh_var "unit" T.unit in
-                    tagE tf.T.lab (unit -->
-                      (primE (Ir.DeserializePrim ts) [varE arg]))
+                  | T.Func(T.Local, _, [], [], ts) ->
+                    tagE tf.T.lab
+                      (funcE ("$"^tf.T.lab) T.Local T.Returns [] [] ts
```
Comment (ggreif) [inline src/lowering/desugar.ml:361]: This might work:
```suggestion
                      T.(funcE ("$"^tf.lab) Local Returns [] [] ts
```

**Review by ggreif — APPROVED** [review]
LGTM!

**Pull request merged**